test/libsolidity/syntaxTests/literalSuffixes/parameters/type_mismatch.sol

```diff
@@ -66,41 +66,41 @@ contract C {
-// TypeError 8838: (1178-1231): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1241-1294): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1304-1357): The type of the literal cannot be converted to the parameter of the suffix function.
+// TypeError 8838: (1124-1141): The literal cannot be converted to type string accepted by the suffix function.
+// TypeError 8838: (1151-1167): The literal cannot be converted to type bytes accepted by the suffix function.
+// TypeError 8838: (1178-1231): The address cannot be converted to type uint256 accepted by the suffix function.
+// TypeError 8838: (1241-1294): The address cannot be converted to type int8 accepted by the suffix function.
+// TypeError 8838: (1304-1357): The address cannot be converted to type bool accepted by the suffix function.
 // TypeError 2505: (1444-1500): Functions that take 2 arguments can only be used as literal suffixes for rational numbers.
-// TypeError 8838: (1510-1565): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1575-1629): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1640-1654): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1664-1678): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1688-1702): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1712-1729): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1777-1793): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1803-1818): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1829-1843): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1853-1867): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1877-1891): The type of the literal cannot be converted to the parameter of the suffix function.
-// TypeError 8838: (1901-1918): The type of the literal cannot be converted to the parameter of the suffix function.
+// TypeError 8838: (1510-1565): The address cannot be converted to type string accepted by the suffix function.
+// TypeError 8838: (1575-1629): The address cannot be converted to type bytes accepted by the suffix function.
+// TypeError 8838: (1640-1654): The number is out of range of type uint256 accepted by the suffix function.
+// TypeError 8838: (1664-1678): The number is out of range of type int8 accepted by the suffix function.
```

Comment:
This one still needs tweaking. It's this bit:
```solidity
1.1 uintSuffix;
1.1 int8Suffix;
```

Here it's not about the range. It's that the fractional number cannot be converted into an integer. It would be fine to stick to the generic wording in this case: `The literal cannot be converted to type ...`.

Reply:
fixed in [cf960ef](https://github.com/ethereum/solidity/pull/14015/commits/cf969efb57cc61c52e08d7f7b3a7acfc534224b8).